docs/tutorials/documenting-api-endpoints.md

```diff
@@ -0,0 +1,121 @@
+# Documenting REST API endpoints
+
+This document briefly explains how to document Zulip's REST API endpoints
+on our `/api` page.
+
+Our API documentation files live under `templates/zerver/api/*`. To begin,
+we recommend copying over an existing doc file (`render-message.md` is a good
+example) and using it as a template. Make sure you link to your new Markdown
+file in `templates/zerver/help/rest-endpoints.md`, so that it can be displayed
+in the sidebar on the `/api` page.
+
+## Usage examples.
+
+We display usage examples in three languages: Python, JavaScript and `curl`.
+For JavaScript and `curl` we simply recommend copying and pasting the examples
+directly into the Markdown file. However, since Zulip's Python bindings are
+used most frequently, the process of adding Python examples for an endpoint
+is a bit more involved.
+
+We recommend skimming `zerver/lib/api_test_helpers.py` before proceeding with the
+steps below.
+
+1. Add a function for the endpoint you'd like to document to
+   `zerver/lib/api_test_helpers.py`. `render_message` is a good example to
+   follow.
+
+1. Add the function to the `TEST_FUNCTIONS` dict.
+
+1. Make the desired API call inside the function. If our Python bindings don't
+   have a dedicated method for a specific API call, you may either use
+   `client.call_endpoint` or add a dedicated function to the
+   [`zulip` package](https://github.com/zulip/python-zulip-api/tree/master/zulip).
```

Comment:
Not sure if this part is formatted as intended.

**Screenshot (both zulip and package are clickable):**
![image](https://user-images.githubusercontent.com/21148168/40938139-c444d200-67f5-11e8-8a83-2b1a223b7b6c.png)